packages/@sanity/portable-text-editor/src/types/slate.ts

```diff
@@ -0,0 +1,21 @@
+import {BaseEditor, Descendant} from 'slate'
+import {ReactEditor} from '@sanity/slate-react'
+import {PortableTextSlateEditor, TextBlock, TextSpan} from '..'
+
+interface VoidElement {
+  _type: string
+  _key: string
+  children: Descendant[]
+  __inline: boolean
+  value: {
+    [other: string]: any
+  }
+}
+
+declare module 'slate' {
+  interface CustomTypes {
+    Editor: BaseEditor & ReactEditor & PortableTextSlateEditor
+    Element: TextBlock | VoidElement
+    Text: TextSpan
+  }
+}
```

Comment:
Is this the official way to extend Slate types? Can we add a link to the docs here?